tensorflow/lite/micro/tools/make/targets/cortex_m_generic_makefile.inc

```diff
@@ -1,156 +1,145 @@
+  CCFLAGS += $(FLAGS_ARMC)
+  LDFLAGS += $(ARM_LDFLAGS)
 
-    # Arm Compiler will not link the Math library (see below), therefore we're filtering it out.
-    # See Fatal error: L6450U: Cannot find library m:
-    # "Arm Compiler is designed to run in a bare metal environment,
-    # and automatically includes implementations of these functions,
-    # and so no such flag is necessary."
-    # https://developer.arm.com/documentation/100891/0611/troubleshooting/general-troubleshooting-advice
-    MICROLITE_LIBS := $(filter-out -lm,$(MICROLITE_LIBS))
-  else ifeq ($(filter armgcc,$(ALL_TAGS)),armgcc)
-    ifeq ($(shell which arm-none-eabi-gcc),)
-      export PATH := $(MAKEFILE_DIR)/downloads/gcc_embedded/bin/:$(PATH)
-      $(eval $(call add_third_party_download,$(GCC_EMBEDDED_URL),$(GCC_EMBEDDED_MD5),gcc_embedded,))
-    endif
+  # Arm Compiler will not link the Math library (see below), therefore we're filtering it out.
+  # See Fatal error: L6450U: Cannot find library m:
+  # "Arm Compiler is designed to run in a bare metal environment,
+  # and automatically includes implementations of these functions,
+  # and so no such flag is necessary."
+  # https://developer.arm.com/documentation/100891/0611/troubleshooting/general-troubleshooting-advice
+  MICROLITE_LIBS := $(filter-out -lm,$(MICROLITE_LIBS))
+else ifeq ($(TOOLCHAIN), armgcc))
+  ifeq ($(shell which arm-none-eabi-gcc),)
+    export PATH := $(MAKEFILE_DIR)/downloads/gcc_embedded/bin/:$(PATH)
+    $(eval $(call add_third_party_download,$(GCC_EMBEDDED_URL),$(GCC_EMBEDDED_MD5),gcc_embedded,))
+  endif
 
-    CXX_TOOL  := arm-none-eabi-gcc
-    CC_TOOL   := arm-none-eabi-gcc
-    AR_TOOL   := arm-none-eabi-gcc-ar
-    LD        := arm-none-eabi-ld
-    ifneq ($(filter cortex-m55%,$(GCC_TARGET_ARCH)),)
-      $(error Micro architecure support is not available yet for $(GCC_TARGET_ARCH))
-    else
-      FLAGS_GCC = -mcpu=$(GCC_TARGET_ARCH)
-    endif
-    CXXFLAGS += $(FLAGS_GCC)
-    CCFLAGS += $(FLAGS_GCC)
-    LDFLAGS +=
+  TARGET_TOOLCHAIN_PREFIX := arm-none-eabi-
+  
+  ifneq ($(filter cortex-m55%,$(TARGET_ARCH)),)
+    $(error Micro architecure support is not available yet for $(GCC_TARGET_ARCH))
+  else
+    FLAGS_GCC = -mcpu=$(GCC_TARGET_ARCH)
   endif
+  CXXFLAGS += $(FLAGS_GCC)
+  CCFLAGS += $(FLAGS_GCC)
+  LDFLAGS +=
+endif
 
-  PLATFORM_FLAGS = \
-    -DTF_LITE_STATIC_MEMORY \
-    -mthumb \
-    -mfloat-abi=$(FLOAT) \
-    -funsigned-char \
-    -mlittle-endian \
-    -Wno-type-limits \
-    -fno-function-sections \
-    -MD \
-    -DCPU_$(CORE)=1 \
-    $(TARGET_SPECIFIC_FLAGS)
+PLATFORM_FLAGS = \
+  -mthumb \
+  -mfloat-abi=$(FLOAT) \
+  -funsigned-char \
+  -mlittle-endian \
+  -Wno-type-limits \
+  -Wno-unused-private-field \
+  -MD \
+  -DCPU_$(CORE)=1 \
+  $(TARGET_SPECIFIC_FLAGS)
 
-  ifneq ($(BUILD_TYPE), release)
-    PLATFORM_FLAGS += -DTF_LITE_MCU_DEBUG_LOG
-  endif
+ifneq ($(BUILD_TYPE), release)
+  PLATFORM_FLAGS += -DTF_LITE_MCU_DEBUG_LOG
+endif
```

Comment:
```-DT_LITE_MCU_DEBUG_LOG``` should be part of all builds. Else we will have fprintf and abort calls in the library.

https://github.com/tensorflow/tensorflow/blob/b43c3c3a68e4b4e2c7fc4daa76c4f9485c7ed47a/tensorflow/lite/kernels/op_macros.h#L20-L46

Reply:
Ok